### PR TITLE
Followups to #2292, code cleanups for opt-utils.h

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: src/passes/opt-utils.h [hunk 1]
```diff
@@ -56,51 +56,48 @@ inline void optimizeAfterInlining(std::unordered_set<Function*>& funcs,
...
     : replacements(replacements) {}
 
   CallTargetReplacer* create() override {
     return new CallTargetReplacer(replacements);
   }
 
   void visitCall(Call* curr) {
-    auto iter = replacements->find(curr->target);
-    if (iter != replacements->end()) {
+    auto iter = replacements.find(curr->target);
```
Comment (jgravelle-google) [inline src/passes/opt-utils.h]: Could also be maybeReplace'd

File: src/passes/opt-utils.h [hunk 2]
```diff
@@ -56,51 +56,48 @@ inline void optimizeAfterInlining(std::unordered_set<Function*>& funcs,
...
   }
 
 private:
-  std::map<Name, Name>* replacements;
+  const std::map<Name, Name>& replacements;
 };
 
 inline void replaceFunctions(PassRunner* runner,
                              Module& module,
                              std::map<Name, Name>& replacements) {
```
Comment (jgravelle-google) [inline src/passes/opt-utils.h]: Could also be const&'d?

File: src/passes/opt-utils.h (hunk 1 above)
Comment (kripken) [reply src/passes/opt-utils.h]: Done using `std::function` to pass the lambda around... thoughts?

File: src/passes/opt-utils.h (hunk 2 above)
Comment (kripken) [reply src/passes/opt-utils.h]: Done.

File: src/passes/opt-utils.h (hunk 1 above)
Comment (jgravelle-google) [reply src/passes/opt-utils.h]: That's snazzy. It makes `CallTargetReplacer` harder to use with a replacements dictionary (needing to re-build the logic in a lambda elsewhere), but more general (aren't tied to using a dictionary to figure out replacements at all).

**Pull request merged**